File: examples/java/language_gauntlet/src/main/java/gauntlet/runtime/Executor.java
package gauntlet.runtime;

public final class Executor {
    private Executor() {}

    public static String execute(String cmd) {
        try {
            // SINK -- Runtime.exec / CWE-78.
            Runtime.getRuntime().exec(cmd);
        } catch (Exception ignored) {
        }
        return cmd;
    }

    public static String cleanTwin() {
        try {
            // NEGATIVE -- the constant argument must remain untainted.
            Runtime.getRuntime().exec("echo clean");
        } catch (Exception ignored) {
        }
        return "clean";
    }
}

File: examples/go/language_gauntlet/internal/runtime/executor.go
package runtime

import execpkg "os/exec"

func Execute(cmd string) int {
	// SINK -- shell wrapper with a tainted command / CWE-78.
	_ = execpkg.Command("/bin/sh", "-c", cmd).Run()
	return 0
}

func CleanTwin() int {
	// NEGATIVE -- the constant argument must remain untainted.
	_ = execpkg.Command("/bin/sh", "-c", "echo clean").Run()
	return 0
}
